pr_agent/algo/utils.py

```diff
@@ -1,5 +1,8 @@
 from __future__ import annotations
 
+import json
+import logging
+import re
 import textwrap
 
```

Comment:
Consider adding a comment explaining why the maximum number of iterations is set to 10. This would improve code readability and maintainability. [medium]